web/src/search_suggestion.ts

```diff
@@ -6,26 +6,50 @@ import * as huddle_data from "./huddle_data";
 import * as narrow_state from "./narrow_state";
 import {page_params} from "./page_params";
 import * as people from "./people";
+import type {User} from "./people";
+import type {NarrowTerm} from "./state_data";
 import * as stream_data from "./stream_data";
 import * as stream_topic_history from "./stream_topic_history";
 import * as stream_topic_history_util from "./stream_topic_history_util";
 import * as typeahead_helper from "./typeahead_helper";
 
+type UserPillItem = {
+    id: number;
+    display_value: Handlebars.SafeString;
+    has_image: boolean;
+    img_src: string;
+    should_add_guest_user_indicator: boolean;
+};
+
+type InvalidItem = Omit<NarrowTerm, "operand"> & Partial<Pick<NarrowTerm, "operand">>;
+
+type Suggestion = {
+    description_html: string;
+    search_string: string;
+    invalid?: InvalidItem[];
+    is_person?: boolean;
+    user_pill_context?: UserPillItem;
+};
+
 export const max_num_of_search_results = 12;
 
-function stream_matches_query(stream_name, q) {
+const noop = (): void => {
+    // This function does nothing
+};
```

Comment:
It would be better to just have this explicitly done in the one place we're using it, and have the comment instead explain __why__ we're not doing anything.